File: agenkit-java/src/main/java/io/agenkit/evaluation/Metric.java
package io.agenkit.evaluation;

/**
 * A named evaluation metric with a score and optional metadata.
 */
public record Metric(String name, double score, String description) {

    public Metric(String name, double score) {
        this(name, score, "");
    }

    public boolean passes(double threshold) {
        return score >= threshold;
    }

    @Override
    public String toString() {
        return "Metric{" + name + "=" + String.format("%.3f", score) + "}";
    }
}

File: agenkit-cs/src/Agenkit/Evaluation/Metric.cs
namespace Agenkit.Evaluation;

/// <summary>
/// A single evaluation metric with name, value, and unit.
/// </summary>
public record Metric(string Name, double Value, string Unit = "")
{
    /// <summary>Returns a human-readable representation.</summary>
    public override string ToString() =>
        string.IsNullOrEmpty(Unit) ? $"{Name}: {Value}" : $"{Name}: {Value} {Unit}";
}
